test/utils/fnIt.ts

```diff
@@ -0,0 +1,16 @@
+import { spy } from "ts-mockito";
+
+const getName = <T>(fn: (arg: T) => (...args: any[]) => any) => {
+    // create an empty instance
+    const dummyInstance = {} as T;
+    // spy it to mark up some properties
+    const dummySpy: T = spy(dummyInstance);
+    // used this 'typed' spy to extract the name
+    return fn(dummySpy).name;
+};
+
+/* fnIt can be used to retrieve function names rather than writing them manually. This will be useful when refactoring
```

Comment:
You put the comment in the js doc style:
```
/**
 *
 */
```
If you do that it'll show up when someone hovers over `fnIt` in any file. Also I like the second part of the comment, but I dont think the first part is true: getName can be used to retrieve function names, fnIt cant. I would write something like:
`A test fixture for a specific function. The supplied function will be used to name the test. This will be useful when since refactoring there is no need to go back an change all the tests manually in case a function name has been changed. Types of T that are generic can be declared with 'any' eg. fnIt<BlockCache<any>>(b => b.setHead, ...)`